### DISCUSSION TITLE
Quorum Queues Message Count Stats

### DISCUSSION THREAD

**Discussion by jerome-zenfolio** [rabbitmq/rabbitmq-server#3235] [Questions]
Hello,

The message count stats for a quorum queue that is observed on RabbitMQ management UI and  HTTP API are showing incorrect values. I was able to reproduce the scenario in:
```
RabbitMQ 3.7.8 Erlang 21.1
RabbitMQ 3.8.9 Erlang 22.3.4.11
```
In my case a consumer starts processing messages from a queue of say 40 messages. As each message is acknowledged, I would expect to see the message count decreasing in the Management UI. However when observing the UI, the message count is not updated until all 40 messages are acknowledged.

![quorum](https://user-images.githubusercontent.com/40347213/127069206-44eacd95-0d1d-48a3-bf79-6d986c14485e.JPG)

Where as when I use a classic queue for the exact same app, I see the expected result in the UI:

![classic](https://user-images.githubusercontent.com/40347213/127069300-e442d69c-1b58-4f80-961b-0416a38c515c.JPG)

Monitoring the size of a queue and alerting are a critical part of our infrastructure. Could you point me in the direction on how to accurately get these stats for quorum queues?

Thanks!

**Comment by kjnilsson** [comment]
I wrote a test program that consumed from a quorum queue with 40 messages acking 1 message every 5 seconds and I get this graph. Please can you try the latest version of RabbitMQ (3.8.9 is quite old now). If that still shows the issue then there must be something else going on either with you consumer (acking all at once) or you configuration.

<img width="618" alt="Screenshot 2021-07-27 at 09 14 27" src="https://user-images.githubusercontent.com/1180564/127120373-9bcf184d-e7d4-4063-9799-25a6025fcdce.png">

**Reply by jerome-zenfolio** [reply]
Karl, thank you for the quick response and taking the time to test it, it turns out the issue was on our workflow which is identified and corrected.
